Sources/SmithyTestUtil/RequestTestUtil/HttpRequestTestBase.swift

```diff
@@ -237,6 +264,17 @@ open class HttpRequestTestBase: XCTestCase {
     ) async throws {
         let expectedData = try await expected.readData()
         let actualData = try await actual.readData()
+
+        // Unseekable streams may have already been read by Signer middleware and cannot be read again
+        // Compare stream lengths if ByteStream is a .stream and actualData is nil
+        if checkStreamLengthEqualityIfUnseekableActualStreamHasBeenRead(
+            expected: expected, actual: actual, file: file, line: line
+        ) {
+            // Stream lengths were checked and comparing data will result in failure due to above conditions
+            return
+        }
```

Comment:
There is an edge case where the SignerMiddleware has already read an unseekable stream. In which case by the time all middlewares are executed we cannot re-read the data. In that case let's check if the actual and expected stream lengths are the same.

Note: This comparison should only fail if the actualStream.position == actualStream.length and !actualStream.isSeekable